library/src/blas3/rocblas_trsm.hpp

```diff
@@ -86,6 +86,35 @@ namespace
                            offset_dst);
     }
 
+    // copy helper for batched arrays of pointers
+    template <typename T>
+    __global__ void setup_batched_array_kernel(T*             src,
+                                               rocblas_stride src_stride,
+                                               T*             dst[])
+    {
+        dst[hipBlockIdx_x] = src + hipBlockIdx_x * src_stride;
+    }
+
+    template <rocblas_int BLOCK, typename T>
+    void setup_batched_array(rocblas_handle handle,
+                             T*             src,
+                             rocblas_stride src_stride,
+                             T*             dst[],
+                             rocblas_int    batch_count)
+    {
+        dim3 grid(batch_count);
+        dim3 threads(BLOCK);
+
+        hipLaunchKernelGGL(setup_batched_array_kernel<T>,
+                           grid,
+                           threads,
+                           0,
+                           handle->rocblas_stream,
+                           src,
+                           src_stride,
+                           dst);
+    }
+
```

Comment:
Maybe this should be moved to a common file to be used with trsv

Reply:
I moved it to the trtri_trsm.hpp file in 11313d6. If anyone thinks it should be moved somewhere else let me know.